eth_typing/abi.py

```diff
@@ -54,7 +54,7 @@ class ABIEventParam(TypedDict, total=False):
     """Type of the event parameter."""
 
 
-class ABIComponent(TypedDict):
```

Comment:
If I'm understanding correctly, since we only need components to be optional, could we use `NotRequired` instead of using `total=False`? So `components: NotRequired[Sequence["ABIComponent"]]`

Reply:
Looks like you may need to import both TypedDict and NotRequired from typing_extensions if you go that route

Reply:
Good callout, I didn't know about this feature!

Basically all types were created with "optional" attributes using `total=False`. I think implementing them correctly so that the fields that are really required can be checked appropriately. I will tackle that in a new PR.

Reply:
Decided the change wasn't that significant so made those updates here.

Reply:
haha sounds good. Approved!